src/TLuaInterpreter.cpp

```diff
@@ -7094,6 +7137,157 @@ int TLuaInterpreter::permSubstringTrigger( lua_State * L )
     return 1;
 }
 
+int TLuaInterpreter::permKey( lua_State *L )
+{
+    string luaNameID = "";
+    if( ! lua_isstring( L, 1 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #1 type (key name as string expected, got %1!)" )
+            .arg( luaL_typename( L, 1 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaNameID = lua_tostring( L, 1 );
+    }
+
+    string luaParent = "";
+    if( ! lua_isstring( L, 2 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #2 type (key parent group as string expected, got %1!)" )
+            .arg( luaL_typename( L, 2 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaParent = lua_tostring( L, 2 );
+    }
+
+    int luaKeyCode;
+    if( ! lua_isstring( L, 3 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #3 type (key code as string expected, got %1!)" )
+            .arg( luaL_typename( L, 3 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaKeyCode = lua_tointeger( L, 3 );
+    }
+
+    int luaModifier;
+    if( ! lua_isstring( L, 4 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #4 type (key modifier as string expected, got %1!)" )
+            .arg( luaL_typename( L, 4 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaModifier = lua_tointeger( L, 4 );
+    }
+
+
+    string luaFunction = "";
+    if( ! lua_isstring( L, 5 ) )
+    {
+        lua_pushstring( L, tr( "permKey: bad argument #5 type (key lua code as string expected, got %1!)" )
+            .arg( luaL_typename( L, 5 ) )
+            .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+    else
+    {
+        luaFunction = lua_tostring( L, 5 );
+    }
+
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "permKey: NULL Host pointer - something is wrong!" )
+            .toUtf8().constData() );
+        return 2;
+    }
+    TLuaInterpreter * pLuaInterpreter = pHost->getLuaInterpreter();
+    QString _luaNameID = luaNameID.c_str();
+    QString _luaParent = luaParent.c_str();
+    QString _luaFunction = luaFunction.c_str();
+    int _luaKeyCode = luaKeyCode;
+    int _luaModifier = luaModifier;
+    int keyID = pLuaInterpreter->startPermKey( _luaNameID, _luaParent, _luaKeyCode, _luaModifier, _luaFunction );
+    lua_pushnumber( L, keyID );
+    return 1;
+}
+
+int TLuaInterpreter::tempKey( lua_State *L )
+{
+    int luaKeyCode;
+    if( ! lua_isstring( L, 1 ) )
+    {
+        lua_pushstring( L, tr( "tempKey: bad argument #1 type (key code as string expected, got %1!)" )
```

Comment:
Hey, sorry to disappear, last weekend got crazy moving into a new house, and this weekend I'm on a trip to Canada. Working on getting the changes you suggested made now, might have them submitted tonight, but I'll let you know. Depends on if my vnc client lets me code remotely from my laptop or not.